core/iwasm/fast-jit/fe/jit_emit_control.c

```diff
@@ -384,13 +449,18 @@ handle_op_end(JitCompContext *cc, uint8 **p_frame_ip)
 
         SET_BUILDER_POS(block->basic_block_end);
 
+        /* Pop block and load block results */
+        block = jit_block_stack_pop(&cc->block_stack);
+        if (!load_block_results(cc, block)) {
```

Comment:
L414 `if(!block->incoming_insn_for_end_bb)` also needs to transfer return values from the *if block* or *else block* to the outer block.

Reply:
Seems no need to change the jit_frame, just move result values from popped block's value stack into previous block's value stack?

Reply:
Yep, only return values . No `basic_block`, No modification of `jit_frame`.

Reply:
But still, need to handle the special case of `LABE_TYPE_FUNCTION`

Reply:
```suggestion
        if (block->label_type == LABEL_TYPE_FUNCTION) {
          handle_func_return(cc, block);
          SET_BB_END_BCIP(cc->cur_basic_block, *p_frame_ip - 1);
        }
        else {
          if (!load_block_result(cc, block) {
        }
```